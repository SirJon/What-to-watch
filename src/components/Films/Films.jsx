import React, { PureComponent } from "react";
import PropTypes from 'prop-types';
import VideoPlayer from "../VideoPlayer/VideoPlayer.jsx";

class Films extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      films: `undefined`,
    };
  }

  onMouseOver = (name) => {
    this.setState((prevState) => ({
      ...prevState,
      films: `${name}`,
    }));
  };

  render() {
    return (
      <div className="catalog__movies-list">
        {/* {this.props.onFilmsGet().films.map((filmsData, i) => {
          const { name, img, video } = filmsData;
          return (
            <VideoPlayer
              key={name + i}
              name={name}
              img={img}
              video={video}
              onMouse={this.onMouseOver}
            />
          );
        })} */}
        {this.props.films.map((filmsData, i) => {
          const { name, img, video } = filmsData;
          return (
            <VideoPlayer
              key={name + i}
              name={name}
              img={img}
              video={video}
              onMouse={this.onMouseOver}
            />
          );
        })}
      </div>
    );
  }
};

Films.propTypes = {
  films: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    img: PropTypes.string,
    video: PropTypes.string,
    genre: PropTypes.string,
  })).isRequired,
};

export default Films;
