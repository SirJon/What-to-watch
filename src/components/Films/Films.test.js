import renderer from 'react-test-renderer';
import React from "react";
import Films from './Films';

const films = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  },
  {
    name: `Bohemian Rhapsody`,
    img: `img/bohemian-rhapsody.jpg`,
  },
  {
    name: `Macbeth`,
    img: `img/macbeth.jpg`,
  },
  {
    name: `Aviator`,
    img: `img/aviator.jpg`,
  },
];

describe(`Films`, () => {
  it(`Render`, () => {
    const tree = renderer
      .create(
        <Films
          films={films}
        />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
