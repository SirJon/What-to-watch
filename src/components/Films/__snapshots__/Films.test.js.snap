// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Films Render 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted="muted"
        poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      >
        <source />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted="muted"
        poster="img/bohemian-rhapsody.jpg"
        width="280"
      >
        <source />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted="muted"
        poster="img/macbeth.jpg"
        width="280"
      >
        <source />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
      >
        Macbeth
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted="muted"
        poster="img/aviator.jpg"
        width="280"
      >
        <source />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
        onClick={[Function]}
      >
        Aviator
      </a>
    </h3>
  </article>
</div>
`;
